pkg/utils/common/controller.go

```diff
@@ -53,3 +60,111 @@ func (s *SuiteController) CheckIfClusterTaskExists(name string) bool {
 	}
 	return false
 }
+
+// Check if a secret exists, return secret and error
+func (s *SuiteController) VerifySecretExists(ns string, name string) (*corev1.Secret, error) {
+	secret, err := s.KubeInterface().CoreV1().Secrets(ns).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return secret, nil
```

Comment:
Perhaps a bit simpler?

```suggestion
	return s.KubeInterface().CoreV1().Secrets(ns).Get(context.TODO(), name, metav1.GetOptions{})
```

Reply:
absolutely simpler. I'll update all of them